ant/src/process.lua

```diff
@@ -0,0 +1,175 @@
+-- Adjust package.path to include the current directory
+-- utils
+local json = require(".json")
+local utils = require(".utils")
+local errors = require(".errors")
+local constants = require(".constants")
+
+-- core modules
+local balances = require(".balances")
+local initialize = require(".initialize")
+local records = require(".records")
+local controllers = require(".controllers")
+
+Name = Name or "Arweave Name Token"
+Ticker = Ticker or "ANT"
+Logo = Logo or "LOGO"
+Denomination = Denomination or 1
+
+local ActionMap = {
+	-- write
+	SetController = "SetController",
+	RemoveController = "RemoveController",
+	SetRecord = "SetRecord",
+	RemoveRecord = "RemoveRecord",
+	SetName = "SetName",
+	SetTicker = "SetTicker",
+	--- initialization method for bootstrapping the contract from other platforms ---
+	InitializeState = "InitializeState",
+	-- read
+	GetControllers = "GetControllers",
+	GetRecord = "GetRecord",
+	GetRecords = "GetRecords",
+}
+
+local TokenSpecActionMap = {
+	Info = "Info",
+	Balances = "Balances",
+	Balance = "Balance",
+	Mint = "Mint",
+	Transfer = "Transfer",
+	CreditNotice = "Credit-Notice",
+}
+
+-- Handlers for contract functions
+-- TokenSpecActionMap
+Handlers.add(TokenSpecActionMap.Transfer, utils.hasMatchingTag("Action", TokenSpecActionMap.Transfer), function(msg)
+	local recipient = msg.Tags.Recipient
+	local function checkAssertions()
+		assert(recipient, "Recipient is required")
+		assert(utils.validateArweaveId(recipient), "Invalid recipient")
+		assert(Balances[msg.From], "Sender is not the owner")
```

Comment:
The token should be transferrable if msg.From is the Owner, has a balance or is the env.ProcessId.

Reply:
Switched to use the validation method and updated the method